src/Features/Core/Portable/ChangeSignature/ChangeSignatureResult.cs

```diff
@@ -12,19 +12,30 @@ internal sealed class ChangeSignatureResult
         public Solution? UpdatedSolution { get; }
         public Glyph? Glyph { get; }
         public bool PreviewChanges { get; }
+        public CannotChangeSignatureReason? CannotChangeSignatureReason { get; }
+        public string? ConfirmationMessage { get; }
 
         /// <summary>
         /// Name of the symbol. Needed here for the Preview Changes dialog.
         /// </summary>
         public string? Name { get; }
 
-        public ChangeSignatureResult(bool succeeded, Solution? updatedSolution = null, string? name = null, Glyph? glyph = null, bool previewChanges = false)
+        public ChangeSignatureResult(
+            bool succeeded,
+            Solution? updatedSolution = null,
+            string? name = null,
+            Glyph? glyph = null,
+            bool previewChanges = false,
+            CannotChangeSignatureReason? cannotChangeSignatureReason = null,
```

Comment:
fwiw i think ChangeSignatureFailureReason or ChangeSignatureFailureKind or ChangeSignatureFailureREsult are better names for `Cannot...Reason`.